### ISSUE TITLE
Google OAuth Token keeps expiring, what to do against it?

### ISSUE THREAD

**Issue by Chaos02** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#595]
Hi, it's pretty much in the title:
After a couple days of using and not using it the OAuth Token needs to be reset.
While this process is fairly simple it is annoying. What can I do to circumvent the token from expiring??

**Comment by Myoldmopar** [comment]
I have the same issue, mine keep expiring

**Comment by momo832** [comment]
Is this expected behavior?

**Comment by im-coder-lg** [comment]
Yeah it is.

**Comment by MichaelDavidGK** [comment]
If app is set as published in google console you can set OAuth Tokens with longer expiry date.
Would need to be changed in code as well ;)

**Comment by BierDav** [comment]
Is there another authentication method which would work too?

**Comment by rdsroka2** [comment]
Interested in a solution to this as well. Will wait for someone smarter than me to come along...

**Comment by MichaelDavidGK** [comment]
> Is there another authentication method which would work too? And I have not dealt with that yet, but does the OAuth Token flow have a way of refreshing the tokens. So that the old one gets invalid and you get a new one with the help of a refresh token and the client id.
> 
> https://cloud.google.com/apigee/docs/api-platform/antipatterns/oauth-long-expiration
> 
> They say that you can have the refresh token can last as long as 200 days

As I said above if you publish the app you get no expiry.
Apparently after the program set expiration date (7 days) it still didn't ask me for new Token (most likely due to the code not forcing it after set expiration).
On top of that it set expiration to Sunday, December 5, 53954 4:41:07am after it was set to expire before (I got token at 5th Dec 2021 4:41 am).

Coming to conclusion:
Simply publishing app in Google Console and create new Token after that and it won't expire.

**Comment by BierDav** [comment]
Would be kind if you offer your published app, because you have already published it and it would simplify a lot.

**Comment by johnwang16** [comment]
> 

I believe what @MichaelDavidGK is saying is you set the app to published in your own google console. 
![image](https://user-images.githubusercontent.com/47154554/149674072-8b128945-5056-4524-97e8-86f59ba86898.png)

**Comment by BierDav** [comment]
Yes I'm aware of that, but it would be far more efficient when there is just one Google Cloud Platform Application. Now everybody has to make an own one. And now everybody should publish his app. When it's published it is automatically accessible to everybody why haven hundreds of Applications when just one of them would work too. This would also reduce the setup time, because you don't need to create your own Application

**Comment by MichaelDavidGK** [comment]
There are limited API requests, user cap and other things, might also be security issue, so people would have to do it themselves anyway

**Comment by momo832** [comment]
> If app is set as published in google console you can set OAuth Tokens with longer expiry date. Would need to be changed in code as well (it probably sets expiry in code since my token has been showing as expired in this app, even tho it worked in another) ;)
> 
> Otherwise it's only 7 days

Don't you need a domain for this? I tried one and it can't find my TXT or CNMAE record...

**Comment by MichaelDavidGK** [comment]
![image](https://user-images.githubusercontent.com/39009683/149682966-f024ee0a-25e4-43c0-935d-f9ab1ee488a4.png)
nope, mine is without any

**Comment by momo832** [comment]
![image](https://user-images.githubusercontent.com/67990067/149683156-50e61ee5-f41d-4cfc-91f1-ef211876a1ae.png)
I hope this means it should still work.

**Comment by MichaelDavidGK** [comment]
Don't really know, it didn't ask me to add or verify my domain at all

**Comment by BierDav** [comment]
> There are limited API requests, user cap and other things, might also be security issue, so people would have to do it themselves anyway

Oh I didn't know that, but that makes sense

**Comment by cmt29** [comment]
Does anyone know roughly how long it should take after you've pressed Publish App?
